src/nano_contracts/deserializer.ts

```diff
@@ -192,53 +257,85 @@ class Deserializer {
    * The signedData what will be deserialized is
    * [len(serializedValue)][serializedValue][inputData]
    *
-   * @param {signedData} Buffer with serialized signed value
-   * @param {type} Type of the signed value, with the subtype, e.g., SignedData[str]
+   * @param signedData Buffer with serialized signed value
+   * @param type Type of the signed value, with the subtype, e.g., SignedData[str]
    *
    * @memberof Deserializer
    * @inner
    */
-  toSigned(signedData: Buffer, type: string): string {
-    // Get signed data type inside []
-    const match = type.match(/\[(.*?)\]/);
-    const valueType = match ? match[1] : null;
-    if (!valueType) {
+  toSigned(signedData: Buffer, type: string): DeserializeResult<string> {
+    const [containerType, internalType] = getContainerInternalType(type);
+    if (containerType !== 'SignedData') {
+      throw new Error('Type is not SignedData');
+    }
+    if (!internalType) {
```

Comment:
This will throw before, when calling `getContainerInternalType`. But I don't mind keeping this throw here as well.

Reply:
This method was fixed on the other PR, maybe we should address any necessary changes there.